src/main/java/com/aws/greengrass/shadowmanager/sync/SyncConfigurationUpdater.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.shadowmanager.sync;
+
+import com.aws.greengrass.config.Node;
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.config.UpdateBehaviorTree;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.shadowmanager.model.configuration.ThingShadowSyncConfiguration;
+import lombok.Setter;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+import static com.aws.greengrass.componentmanager.KernelConfigResolver.CONFIGURATION_CONFIG_KEY;
+import static com.aws.greengrass.shadowmanager.model.Constants.CONFIGURATION_ADDED_ON_INTERACTION_TOPIC;
+import static com.aws.greengrass.shadowmanager.model.Constants.CONFIGURATION_SYNCHRONIZATION_TOPIC;
+
+public class SyncConfigurationUpdater {
```

Comment:
Can you create this file and rename to match the ones in here? https://github.com/aws-greengrass/aws-greengrass-shadow-manager/tree/main/src/main/java/com/aws/greengrass/shadowmanager/configuration

You can use this to get, update or process the config under synchronize/addOnInteraction keys. That means, the changes in `ShadowSyncConfiguration` file can be moved here.